spec/controllers/users_controller_spec.rb

```diff
@@ -38,6 +38,31 @@
         expect(response).to render_template(:new)
       end
     end
+
+    context "welcome email" do
+      after { ActionMailer::Base.deliveries.clear }
+
+      it "sends out the email to new user when credentials are valid" do
+        post :create, user: Fabricate.attributes_for(:user)
+        expect(ActionMailer::Base.deliveries).to_not be_empty
+      end
+
+      it "sends the email to new user when credentials are valid" do
+        post :create, user: Fabricate.attributes_for(:user, email: "joe@doe.com")
+        email_message = ActionMailer::Base.deliveries.last
```

Comment:
Where are you using `email_message` ? in the expectation 1 line below, you go for `ActionMailer::Base.deliveries.last`